### PR TITLE
Add dragonbox-cmake

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add our own CMakeList for dragonbox which was added  in #17831

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@nikitamikhaylov 

> The source directory /home/jakalletti/ClickHouse/ClickHouse/contrib/dragonbox does not contain a CMakeLists.txt file.

But the file exists and CI successfully builds the project.

**Comment by alexey-milovidov** [comment]
Nevertheless, we tend to use our own CMakeLists for libraries.

File: contrib/dragonbox-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+set(DRAGONBOX_INCLUDE_DIR ${ClickHouse_SOURCE_DIR}/contrib/dragonbox/include)
+
+add_library(dragonbox INTERFACE)
```
Comment (alexey-milovidov) [inline contrib/dragonbox-cmake/CMakeLists.txt:3]: But there is cpp file to compile.

Comment (nikitamikhaylov) [reply contrib/dragonbox-cmake/CMakeLists.txt:3]: We use only `dragonbox/dragonbox_to_chars.h` in our codebase, I thought that it is enough.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**